inbm/genai_unit_tests/ubuntu_source_manager_genai_unit_test.py

```diff
@@ -0,0 +1,339 @@
+import sys
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/dispatcher/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/dispatcher/source/')
+sys.path.append('/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm-lib/')
+import pytest
+from unittest.mock import patch, MagicMock
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, ApplicationAddSourceParameters
+from unittest.mock import patch, mock_open
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, ApplicationSourceList
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, ApplicationRemoveSourceParameters, SourceError
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, SourceError, ApplicationUpdateSourceParameters
+from dispatcher.source.ubuntu_source_manager import UbuntuOsSourceManager, SourceParameters, SourceError
+from dispatcher.source.ubuntu_source_manager import UbuntuOsSourceManager, SourceError
+from dispatcher.source.ubuntu_source_manager import UbuntuApplicationSourceManager, SourceError, ApplicationRemoveSourceParameters
+
+sys.path.insert(0, '../')
+sys.path.insert(0, '../../../')
+sys.path.insert(0, '../../')
+sys.path.insert(0, '../../../../')
+sys.path.insert(0, './')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/intel-inb-manageability/intel-inb-manageability/inbm/dispatcher-agent/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/intel-inb-manageability/intel-inb-manageability/inbm-lib/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/intel-inb-manageability/intel-inb-manageability/inbm/dispatcher-agent/dispatcher/source/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm-lib/')
+sys.path.insert(0, '/home/runner/GITHUB_ACTION_RUNNERS/_work/genai_ag_intel-inb-manageability/genai_ag_intel-inb-manageability/inbm/dispatcher-agent/dispatcher/source/')
+#DO NOT DELETE THIS LINE - TestUbuntuApplicationSourceManagerList
+'''
+ADD HUMAN FEEDBACK BELOW:
+
+'''
+class TestUbuntuApplicationSourceManagerList:
```

Comment:
To go back to Pull Request Summary table, Click on the [link](https://github.com/ankit-r-gajera/genai_ag_intel-inb-manageability/pull/7#)

<h4>For feedback purposes, please include both emoticon-based and qualitative feedback in your reply to this comment.</h4>

To go back to Pull Request Summary table, Click on the [link](https://github.com/ankit-r-gajera/genai_ag_intel-inb-manageability/pull/7#)